teuthology/orchestra/daemon/group.py

```diff
@@ -57,13 +59,16 @@ def register_daemon(self, remote, type_, id_, *args, **kwargs):
         if id_ in self.daemons[role]:
             self.daemons[role][id_].stop()
             self.daemons[role][id_] = None
+
         klass = DaemonState
-        if remote.init_system == 'systemd':
+        if self.use_ceph_daemon:
+            klass = CephDaemonUnit
+        elif self.use_systemd and \
+             not any(map(lambda i: i == 'valgrind', args)) and \
```

Comment:
nit: for concise and better readability it is enough ```not any(i == 'valgrind' for i in args)```

Reply:
fixed